src/clj/fluree/db/connection.cljc

```diff
@@ -139,89 +149,157 @@
   [conn ledger-alias]
   (go-try
     (if (relative-ledger-alias? ledger-alias)
-      (let [nameservices (all-nameservices conn)]
-        (loop [nameservices* nameservices
-               addresses     []]
-          (let [ns (first nameservices*)]
-            (if ns
-              (if-let [address (<? (nameservice/address ns ledger-alias))]
-                (recur (rest nameservices*) (conj addresses address))
-                (recur (rest nameservices*) addresses))
-              addresses))))
+      (loop [nameservices* (publishers conn)
+             addresses     []]
+        (let [ns (first nameservices*)]
+          (if ns
+            (if-let [address (<? (nameservice/publishing-address ns ledger-alias))]
+              (recur (rest nameservices*) (conj addresses address))
+              (recur (rest nameservices*) addresses))
+            addresses)))
       [ledger-alias])))
 
 (defn primary-address
   "From a connection, lookup primary address from nameservice(s) for a given
   ledger alias"
-  ([conn ledger-alias]
-   (go-try
-     (first (<? (addresses conn ledger-alias))))))
+  [{:keys [primary-publisher] :as _conn} ledger-alias]
+  (nameservice/publishing-address primary-publisher ledger-alias))
 
-(defn lookup-commit
+(defn lookup-commit*
   "Returns commit address from first matching nameservice on a conn
    for a given ledger alias and branch"
+  [ledger-address nameservices]
+  (go-try
+    (loop [nses nameservices]
+      (when-let [nameservice (first nses)]
+        (or (<? (nameservice/lookup nameservice ledger-address))
```

Comment:
Do we have any way to ensure that this address is the most up-to-date? Say we have two nameservices, one is offline for quite a while but then comes back on right as we load, will we fork at that point? And will we have a mechanism for "rebasing" if we need to later?

Reply:
After discussing, this will create a fork. Also, if the only nameservice with a ledger is read only and the user transacts to it, a local fork will be created.